trunner/harness/base.py

```diff
@@ -92,9 +106,14 @@ def _reboot_hard(self):
         self.host.set_power(True)
         time.sleep(0.5)
 
-    def _reboot_dut_gpio(self, hard):
+    def _reboot_via_debugger(self, ftdi):
+        pass
+
+    def _reboot_dut_gpio(self, hard, gdb_dev_rst: Optional[bool] = False):
```

Comment:
note: the type is not `Optional[bool]`', just `bool` (`Optional` means that you can pass None, not an optional function parameter)